### PR TITLE
Task #2 Fix: code refactor & events api

### PR DESCRIPTION
**Pull request by suraj-tekdi**
code refactor: 
1. auth/is_user_exist
2. auth/register
3. src/hasura.service.ts --> src/services/hasura.service.ts
4. helper module imported in src/service/hasura.service.ts

events routes add.
events/create api in progress...

auth/register api changes
send sms msg changed

### CODE REVIEW

File: src/src/auth/auth.service.ts [hunk 1]
```diff
@@ -339,7 +339,81 @@ export class AuthService {
 
     }
 
+    public async isUserExist(req: any) {
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:342]: does this validate phone, email, aadhaar all 3?

File: src/src/auth/auth.service.ts [hunk 2]
```diff
@@ -454,4 +528,195 @@ export class AuthService {
...
+        const tableName = 'users';
+        const newR = await this.hasuraService.q(tableName, req, [
+            'first_name',
+            'last_name',
+            'mobile',
+            'email_id',
+            'keycloak_id',
+        ]);
+        const user_id = newR[tableName]?.id;
+        if (user_id) {
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:541]: what about beneficiary role?

File: src/src/auth/auth.service.ts [hunk 3]
```diff
@@ -454,4 +528,195 @@ export class AuthService {
...
+                ...mappedResponse,
+                ['vo_experience']: result?.experience.filter(
+                    (e: any) => e.type == 'vo_experience',
+                ),
+            };
+        }
+
+
+        return {
+            statusCode: 200,
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:716]: statuscode not needed in response

File: src/src/auth/auth.service.ts [hunk 4]
```diff
@@ -339,7 +339,81 @@ export class AuthService {
...
+        const tableName = 'users';
 
+        // Calling hasura common method find all
+        const data_exist = await this.hasuraService.findAll(tableName, req);
+        let response = data_exist.data.users;
+
+        // Check wheather user is exist or not based on response
+        if (response.length > 0) {
+            return {
+                status: 422,
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:353]: statuscode not needed in response

File: src/src/auth/auth.service.ts [hunk 5]
```diff
@@ -339,7 +339,81 @@ export class AuthService {
...
+                message: 'User exist',
+                isUserExist: true,
+            };
+        } else {
+            return {
+                status: 200,
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:359]: statuscode not needed in response

File: src/src/auth/dto/register-facilator.dto.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// src/articles/dto/create-article.dto.ts
```
Comment (manojLondhe) [inline src/src/auth/dto/register-facilator.dto.ts:1]: lets rename this as register

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/auth/auth.service.ts (hunk 1 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.service.ts]: only phone

File: src/src/auth/auth.service.ts (hunk 2 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.service.ts:584]: benificiary role added

File: src/src/auth/auth.service.ts (hunk 5 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.service.ts]: resolved

File: src/src/auth/auth.service.ts (hunk 4 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.service.ts]: resolved

**Pull request merged**